libr/anal/p/anal_mips_gnu.c

```diff
@@ -6,6 +6,732 @@
 #include <r_asm.h>
 #include <r_anal.h>
 
+
+static ut64 t9_pre = UT64_MAX;
+
+// ESIL macros:
+
+// put the sign bit on the stack
+#define ES_IS_NEGATIVE(arg) "1,"arg",<<<,1,&"
+
+#define ES_B(x) "0xff,"x",&"
+#define ES_H(x) "0xffff,"x",&"
+#define ES_W(x) "0xffffffff,"x",&"
```

Comment:
mips.gnu and mips in esil have a lot of things in common, should we put them in separate files?

Reply:
@junchao-loongson you can make a common ".inc" file and include it with the corresponding functions, macroses, definitions, etc.